05-part_two.Rmd

```diff
@@ -4,9 +4,13 @@ From a coding point of view, testing covers the requirements spelled out in the
 
 - console tests 
 - in-script testing
-- unit testing (using tools such as `testthat` in R)
+- formalised testing procedures, such as unit testing (tests the different components or "units" of code using tools such as `testthat` in R)
 
-Testing code within the R console is not recommended as it leaves no trace or records of what has been checked. In-script testing is sufficient to facilitate QA in most cases, although tests should always be accompanied by brief comments signalling their presence and indicating what they are evaluating and what the expected output should be. For a quality assurer, running these tests ensures that the entire analytical workflow is reproducible. In R there are many ways to perfom in-script checks, from `print()` statements, to `base` functions `stop()` and `stopfinot()`, to functions from dedicated packages such as `assertthat` or `testit`. Examples are shown below. 
+## Console tests
+Testing code within the R console is not recommended as it leaves no trace or records of what has been checked.
+
+## In-script testing
+In-script testing is sufficient to facilitate QA in most cases, although tests should always be accompanied by brief comments signalling their presence and indicating what they are evaluating and what the expected output should be. For a quality assurer, running these tests ensures that the entire analytical workflow is reproducible. In R there are many ways to perform in-script checks, from `print()` statements, to `base` functions `stop()` and `stopfinot()`, to functions from dedicated packages such as `assertthat` or `testit`. Examples are shown below. 
```

Comment:
Could we add something about Python and/or other languages here as well?  Even if it's just a simple statement such as: "In R there are........  Equivalent syntyax for Python and other languages is also readily available (any nice links??) but we've included some examples from R below"

Reply:
I'll need to do some research to address this one. I'll see if I can carve out some time later this week.

Reply:
I've upped the Python content now.